guides/multiSigtx-creation-signing/create-multiSig-transaction.js

```diff
@@ -0,0 +1,123 @@
+const { apiClient, cryptography, codec, Transaction, transactions } = require('lisk-sdk');
+const { transferParamsSchema } = require('lisk-framework/dist-node/modules/token/schemas');
+const { accounts } = require('./accounts.json');
+const signedTX = require('./signedTx.json');
+const readline = require("readline");
+const fs = require("fs");
+
+// Change these values as per your node.
+const RPC_ENDPOINT = 'ws://127.0.0.1:7887/rpc-ws';
+const chainID = Buffer.from('00000001', 'hex');
+const tokenID = Buffer.from('0000000100000000', 'hex');
+
+let privateKeyStr;
+let existingSignedTx;
+const optionalAccount1 = accounts[0];
+const optionalAccount2 = accounts[1];
+const optionalAccount3 = accounts[2];
+const sortedOptionalKeys = [Buffer.from(optionalAccount1.publicKey, 'hex'), Buffer.from(optionalAccount2.publicKey, 'hex'), Buffer.from(optionalAccount3.publicKey, 'hex')].sort((a, b) => a.compare(b));
+const sortedMandatoryKeys = [];
+const senderKeyInfo = accounts[0];
+const latestNonce = BigInt(3);
+
+const keys = {
+    mandatoryKeys: sortedMandatoryKeys,
+    optionalKeys: sortedOptionalKeys,
+};
+
+const tokenTransferParams = {
+    tokenID: tokenID,
+    recipientAddress: cryptography.address.getAddressFromLisk32Address(accounts[1].address),
+    amount: BigInt(20000000),
+    data: 'Hello!',
+};
+
+const encodedTransferParams = codec.encode(transferParamsSchema, tokenTransferParams);
+
+const unSignedTx = new Transaction({
+    module: 'token',
+    command: 'transfer',
+    nonce: latestNonce,
+    senderPublicKey: Buffer.from(senderKeyInfo.publicKey, 'hex'),
+    fee: BigInt(1000000),
+    params: tokenTransferParams,
+    signatures: [],
+});
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+
+if (process.argv.length < 3) {
+    console.log("Please provide all the required parameter when executing the script:");
+    console.log("node create-multiSig-transaction-offline.js PRIVATEKEY");
+    process.exit(1);
+}
+
+fs.readFile('signedTx.json', (err, data) => {
```

Comment:
I don't get this part, why do you `require("signedTx.json")` and `readFile("signedTx.json")` at the same time?

Reply:
I require it in case I have acquired a single signature and that signature has been written to the file. I read it in case there aren't any signatures so that I can run the condition accordingly.

Reply:
That is true, however, in the guide I explicitly mention to only add `{}` in the signedTx.json just to ensure that the JSON file structure is followed. If nothing is written in the signedTx.json, the script throws an error whilst requiring this file.